server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/RangeAggregator.java

```diff
@@ -63,6 +67,14 @@
         protected final double to;
         protected final String toAsStr;
 
+        public Range(String key, Double from, String fromAsStr, Double to, String toAsStr) {
```

Comment:
I wonder if we should add a guard here to throw an exception if `from`/`fromAsStr` and `to`/`toAsStr` are both non-null/non-empty/not infinity?  Looks like we previously were just assuming the user gave us the right combination... i'm assuming we use these preferentially later (e.g. `toAsStr` if non-null`, `to` otherwise or something like that)

Or if it turns out to be a big mess, at least a javadoc explaining the priority?  Or make it package-private if possible?  Not sure where else it's used though

Reply:
It is used in a few places, yeah. I believe the best I can do is describe the priority. I'll do that.